Add Conesnsus Exchange Types, and Move into NodeImplementation

bfish713 opened the pull request:
Will close: https://github.com/EspressoSystems/HotShot/issues/927

shenkeyao commented on src/tasks/mod.rs:
```diff
@@ -139,13 +141,45 @@ where
+    let exchange = hotshot.inner.quorum_exchange.clone();
+    let committee_exchange = hotshot.inner.committee_exchange.clone();
+
+    let _network_broadcast_task_handle = async_spawn(
+        network_task(
+            hotshot.clone(),
+            shut_down.clone(),
+            TransmitType::Broadcast,
+            exchange.clone(),
+        )
+        .instrument(info_span!("HotShot Broadcast Task",)),
+    );
+    let _network_direct_task_handle = async_spawn(
+        network_task(
+            hotshot.clone(),
+            shut_down.clone(),
+            TransmitType::Direct,
+            exchange,
+        )
+        .instrument(info_span!("HotShot Direct Task",)),
+    );
```
Just double-checking, do we plan to add these to the `HotShotInner` in the future? Not sure I understand why we add the DA tasks but not these two quorum tasks.

bfish713 replied:
It should be in `TaskHandleInnder` I'll add it there.  It's used for shutdown so I will add it there now.

shenkeyao reviewed (approved):
Looks good! Only minor comments.

bfish713 commented on types/src/certificate.rs:
```diff
@@ -174,13 +177,11 @@ impl<TYPES: NodeType, LEAF: commit::Committable>
-        #[allow(deprecated)]
-        nll_todo()
+    fn leaf_commitment(&self) -> Commitment<TYPES::BlockType> {
+        self.block_commitment
```
Hmm I actually started using this in `is_valid_cert` to make sure the cert commitment is the same as the block commitment from the `DAProposal`

bfish713 commented on src/traits/networking/centralized_web_server_network.rs:
```diff
@@ -46,36 +48,37 @@ use std::{
 #[derive(Clone)]
 pub struct CentralizedWebCommChannel<
     TYPES: NodeType,
+    I: NodeImplementation<TYPES>,
```
Do you mean just here, or everywhere in the code?  It seems like `I: NodeImplementation` was used in a lot of places before this pull request

elliedavidson replied:
I meant everywhere in the code, though I had thought we changed `I` to something else earlier.  But I guess we didn't.  So no need to do it for this PR.  😁

bfish713 replied:
It is `NODE` in one place, I think I'll close this PR and can open another with the rename

shenkeyao commented:
Double-checking a minor thing: I noticed that there were many `error!` to `info!` changes. Were they intentional? Or maybe changed for the convenience of debugging but supposed to be changed back?

shenkeyao reviewed (approved):
LGTM!

bfish713 commented:
Apparently the new version of rust causes a new lint error which I couldn't see locally.  Updated rustc locally and fixed it, but now I need another review :(
